plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -127,15 +129,40 @@ public PluginManager(Config cfg) {
         skipFailedPlugins = cfg.isSkipFailedPlugins();
         hashFunction = cfg.getHashFunction();
         httpClient = null;
+        try{
+            userAgentInformation = this.getUserAgentInformation();
+        }
+        catch(UserAgentInformationException e){
+            logVerbose("Unable to get the UserAgentInformation");
+        }
     }
 
+    private String getUserAgentInformation() throws UserAgentInformationException{
+        Properties properties = new Properties();
+        ClassLoader propertiesClassLoader = this.getClass().getClassLoader();
+        try(InputStream propertiesStream = propertiesClassLoader.getResourceAsStream("Version.properties")){
+            if(propertiesStream != null){
+                properties.load(propertiesStream);
+                userAgentInformation = properties.getProperty("project.artifactId") +"/"+properties.getProperty("project.version");
+            }
+            else{
+                userAgentInformation = "Plugin-Installation-Manager-Tool";
+            }
+        }
+        catch(IOException e){
+            logVerbose("Not able to load/detect Version.properties file");
```

Comment:
this should fall back to the 'dev' version, but code should be refactored so you only need to set the fallback once
```suggestion
            logVerbose("Not able to load/detect Version.properties file");
            userAgentInformation = "Jenkins plugin manager/Dev";
```